### DISCUSSION TITLE
Is clock() actually implemented ?

### DISCUSSION THREAD

**Discussion by GlennFR** [picocomputer/community#122] [Q&A] [answered]
Hi, I'm working on many different // projects on the RP6502, one of which is a port of the PLASMA language.
I wanted to get the clock ticks value (in assembler) but the subroutine doesn't seem to be implemented in any of the std. libraries.
Although cc65/ca65 will correctly compile the call to clock() [or _clock], the linker can't find the actual subroutine to link to.
From the cc65 function reference, this should be a machine specific routine, so aught to be in rp6502.lib...
Any ideas, please?

**Comment by GlennFR** [comment]
I've investigated a bit further.  In the RIA firmware main.c I see this
```
bool main_api(uint8_t operation)
{
    switch (operation)
    {
    case 0x01:
        pix_api_xreg();
        break;
    case 0x02:
        cpu_api_phi2();
        break;
    case 0x03:
        oem_api_codepage();
        break;
    case 0x04:
        rng_api_lrand();
        break;
    case 0x05:
        cpu_api_stdin_opt();
        break;
    case 0x0F:
        clk_api_clock();
        break;
    case 0x10:
        clk_api_get_res();
 ... and so on
```
Note the trap for 0x0F.  But if I try 
```
long get_ticks()
{
    return ria_call_long(0x0F);
}

```
I always get zero.

**Reply by GlennFR** [reply]
Please, anyone ?  Rumble ?

**Reply by rumbledethumps** [reply]
clock() starts counting from 0 when the 6502 starts. Did you wait?

**Reply by GlennFR** [reply]
Hi, yes : the time to type "load clock.rp6502" the counter should have advanced by a few hundred already.  Do you have a piece of test code that I could try, please?

**Reply by rumbledethumps** [reply]
> the time to type "load clock.rp6502" the counter should have advanced by a few hundred already.

This is an incorrect assumption. You didn't wait before reading clock().

**Reply by GlennFR** [reply]
OK, this is my new code, still returns 0.
`static long Ticks;
int     count, temp;

long get_ticks()
{
    return ria_call_long(0x0F);
}

int main(void)
{
    fprintf(stderr, "Please wait...\n");
    fflush(stderr);
    for (count=0; count < 32767; count++)
    {
        temp = temp - count;
    }
    Ticks = get_ticks();
    fprintf(stderr, "Ticks = %u\n", Ticks);
    fprintf(stderr, "Hello.\n");
    return 0;
}`
But my other concern is that clock() doesn't appear in the library rp6502.lib...

**Reply by rumbledethumps** [reply]
%u isn't long. %lu is.

**Reply by GlennFR** [reply]
OK, my error for the format string, thanks.
BUT : rp6502.inc is missing the "0F" code; rp6502.lib is missing the clock() function.

**Reply by rumbledethumps** [reply, marked as answer]
It's not upstream yet but has been in the repo since January:
https://github.com/picocomputer/cc65/commit/2138f3c902f2bcbe538a62fae5e7e36c36ea2c2a
